modules/apps/forms-and-workflow/calendar/calendar-web/src/main/java/com/liferay/calendar/web/internal/portlet/CalendarPortlet.java

```diff
@@ -257,11 +257,34 @@ public void invokeTransition(
 
 		int status = ParamUtil.getInteger(actionRequest, "status");
 
+		long startTime = ParamUtil.getLong(actionRequest, "startTime");
+
+		boolean updateInstance = ParamUtil.getBoolean(
+			actionRequest, "updateInstance");
+
+		boolean allFollowing = ParamUtil.getBoolean(
+			actionRequest, "allFollowing");
+
 		ServiceContext serviceContext = ServiceContextFactory.getInstance(
 			CalendarBooking.class.getName(), actionRequest);
 
-		_calendarBookingService.invokeTransition(
-			calendarBookingId, status, serviceContext);
+		CalendarBooking calendarBooking =
+			_calendarBookingService.invokeTransition(
+				calendarBookingId, status, startTime, updateInstance,
+				allFollowing, serviceContext);
+
+		String redirect = getRedirect(actionRequest, actionResponse);
+
+		if (calendarBooking.getCalendarBookingId() != calendarBookingId) {
+			redirect = StringUtil.replace(
+				redirect, "/-/calendar/event/" + calendarBookingId,
+				"/-/calendar/event/" + calendarBooking.getCalendarBookingId());
```

Comment:
We had problems with some URL manipulations like these. Ideally, it should use HttpUtil or something like that, but AFAIK it is not possible with friendly URLs :( I'm investigating it.